src/Execute/ExecutionManager.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {globalExecutorArray} from '../Backend/Backend';
+import {Device} from './Device';
+
+class ExecutionManager {
+  allDevices: Device[];
+
+  constructor(devices: Device[]) {
+    this.allDevices = devices;
+    for (let device of this.allDevices) {
+      device.checkAvailExecutor(globalExecutorArray);
+    }
+  }
+
+  findDevice(name: string): Device|undefined {
+    return this.allDevices.find(device => {
+      return device.name === name;
+    });
+  }
+}
+
+// This ExecutionManagerMap will be used on Provider.
+interface ExecutionManagerMap {
+  [key: string]: ExecutionManager;
+}
```

Comment:
'used on Provider' means that it will be used somewhere outside? If so, what about moving the interface there? Moreover, as the interface only has a map of ExecutionManager, it'll be more explicit if you defined the map there.

Reply:
I will fix it. Thanks.

Reply:
Could you write a comment explaining what kind of string will be allowed as `key`?

Reply:
> Could you write a comment explaining what kind of string will be allowed as `key`?

Oh I will comment this on next PR update.

Reply:
> Could you write a comment explaining what kind of string will be allowed as `key`?

key will be a location of manager. for example) host on local or IP for some other host.